app/frontend/src/features/auth/jail/Jail.tsx

```diff
@@ -21,29 +20,28 @@ const useStyles = makeStyles((theme) => ({
 export default function Jail() {
   const classes = useStyles(makeStyles);
 
-  const dispatch = useAppDispatch();
-  const isJailed = useTypedSelector((state) => state.auth.jailed);
-  const authError = useTypedSelector((state) => state.auth.error);
-  const authLoading = useTypedSelector((state) => state.auth.loading);
-  const isAuthenticated = useTypedSelector(
-    (state) => state.auth.authenticated === true
-  );
+  const authContext = useAppContext(AuthContext);
+  const isJailed = authContext.jailed;
+  const authError = authContext.error;
+  const authLoading = authContext.loading;
+  const isAuthenticated = authContext.authenticated;
 
   const [loading, setLoading] = useState(false);
   const [jailInfo, setJailInfo] = useState<null | JailInfoRes.AsObject>(null);
 
+  ///TODO: This could fire twice because of authContext dependency
```

Comment:
Hmm I think we would be stuck in an infinite rendering loop because of this since `authContext` will be always updating once this effect runs. Maybe we could solve this by moving this initial check on login success so we don't have to run it here?

Reply:
I might put actions into a different provider.